src/transformState.js

```diff
@@ -5,7 +5,44 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        addProperties(state, extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperties(state, keysToRemove);
+        break;
+
+      case 'clear':
+        clear(state);
+        break;
+
+      default:
+        return;
+    }
+  }
+}
+
+function addProperties(obj, properties) {
+  for (const property in properties) {
+    obj[property] = properties[property];
+  }
+}
```

Comment:
It will be better if you use `Object.assign(target, source)` instead of using a `for...in` loop ✨